add Travis configuration

rjw57 opened the pull request:
This PR adds a Travis configuration to run tests in Djangos 1.8, 1.11 and 2.0 with Pythons 2.7, 3.4, 3.5 and 3.6.

The combination of Django 2.0 and Python 2.7 is excluded since Django 2.0 no longer supports Python 2.7

Commits 3877696 and 0ecaba9 provide fixes for issues with earlier Djangos and Pythons which were found when writing this PR. Commit 9bc63f6 provides a missing mock for the Lookup API which is unavailable outside the CUDN. The remaining test suite was tested with no network connection to make sure there aren't any other unexpected calls to network services.

rjw57 committed:
fix tests for django < 2.0

A cut-and-paste error lead to tests failing on Djangos < 2.0. Fix the
accidental use of "request_user.user" instead of "request_user".

rjw57 committed:
automationcommon.utils: fix for Python 2.7

Fix a use of Python 3-only syntax to be Python 2.7 friendly.

rjw57 committed:
add tox configuration for running tests

rjw57 committed:
add basic travis ci configuration

rjw57 committed:
mock implicit use of lookup in tests

rjw57 committed:
README: add Travis and codecov badges

abrahammartin reviewed (changes requested):
django 1.8 is not compatible with python 3.6: https://docs.djangoproject.com/en/2.0/faq/install/#what-python-version-can-i-use-with-django

rjw57 committed:
travis: exclude Python 3.6 and Django 1.8

Python 3.6 and Django 1.8 is not a supported combination:

https://docs.djangoproject.com/en/2.0/faq/install/#what-python-version-can-i-use-with-django
